node/client/src/lib.rs

```diff
@@ -567,3 +569,252 @@ impl sp_blockchain::HeaderBackend<Block> for Client {
+			$tip:expr,
+			$call:expr,
+			$genesis:expr
+		),
+		{
+			$( $usage:tt )*
+		}
+	} => {
+		match $self {
+			#[cfg(feature = "polkadot")]
+			Self::Polkadot($client) => {
+				use polkadot_runtime as runtime;
+
+				$( $setup )*
+
+				let $extra: runtime::SignedExtra = (
+					frame_system::CheckNonZeroSender::<runtime::Runtime>::new(),
+					frame_system::CheckSpecVersion::<runtime::Runtime>::new(),
+					frame_system::CheckTxVersion::<runtime::Runtime>::new(),
+					frame_system::CheckGenesis::<runtime::Runtime>::new(),
+					frame_system::CheckMortality::<runtime::Runtime>::from(sp_runtime::generic::Era::mortal(
+						$period,
+						$current_block,
+					)),
+					frame_system::CheckNonce::<runtime::Runtime>::from($nonce),
+					frame_system::CheckWeight::<runtime::Runtime>::new(),
+					pallet_transaction_payment::ChargeTransactionPayment::<runtime::Runtime>::from($tip),
+					polkadot_runtime_common::claims::PrevalidateAttests::<runtime::Runtime>::new(),
+				);
+
+				let $raw_payload = runtime::SignedPayload::from_raw(
+					$call.clone(),
+					$extra.clone(),
+					(
+						(),
+						runtime::VERSION.spec_version,
+						runtime::VERSION.transaction_version,
+						$genesis.clone(),
+						$genesis,
+						(),
+						(),
+						(),
+						(),
+					),
+				);
+
+				$( $usage )*
+			},
+			#[cfg(feature = "westend")]
+			Self::Westend($client) => {
+				use westend_runtime as runtime;
+
+				$( $setup )*
+
+				let $extra: runtime::SignedExtra = (
+					frame_system::CheckNonZeroSender::<runtime::Runtime>::new(),
+					frame_system::CheckSpecVersion::<runtime::Runtime>::new(),
+					frame_system::CheckTxVersion::<runtime::Runtime>::new(),
+					frame_system::CheckGenesis::<runtime::Runtime>::new(),
+					frame_system::CheckMortality::<runtime::Runtime>::from(sp_runtime::generic::Era::mortal(
+						$period,
+						$current_block,
+					)),
+					frame_system::CheckNonce::<runtime::Runtime>::from($nonce),
+					frame_system::CheckWeight::<runtime::Runtime>::new(),
+					pallet_transaction_payment::ChargeTransactionPayment::<runtime::Runtime>::from($tip),
+				);
+
+				let $raw_payload = runtime::SignedPayload::from_raw(
+					$call.clone(),
+					$extra.clone(),
+					(
+						(),
+						runtime::VERSION.spec_version,
+						runtime::VERSION.transaction_version,
+						$genesis.clone(),
+						$genesis,
+						(),
+						(),
+						(),
+					),
+				);
+
+				$( $usage )*
+			},
+			#[cfg(feature = "kusama")]
+			Self::Kusama($client) => {
+				use kusama_runtime as runtime;
+
+				$( $setup )*
+
+				let $extra: runtime::SignedExtra = (
+					frame_system::CheckNonZeroSender::<runtime::Runtime>::new(),
+					frame_system::CheckSpecVersion::<runtime::Runtime>::new(),
+					frame_system::CheckTxVersion::<runtime::Runtime>::new(),
+					frame_system::CheckGenesis::<runtime::Runtime>::new(),
+					frame_system::CheckMortality::<runtime::Runtime>::from(sp_runtime::generic::Era::mortal(
+						$period,
+						$current_block,
```

Comment:
Looks like a lot of duplicate code that can continue to be macro-ized?

Reply:
Re: [paritytech/polkadot] Integrate `benchmark-overhead` command (PR #5137)

Reply:
The `SignedExtras` are slightly different for Polkadot, since they have one more entry. But I can sim up the other three ones, which would make the macro a bit more complicated though.